dev-infrastructure/modules/aks-cluster-base.bicep

```
@@ -469,6 +469,36 @@ resource uami_fedcred 'Microsoft.ManagedIdentity/userAssignedIdentities/federate
   }
 ]
 
+resource pullerIdentity 'Microsoft.ManagedIdentity/userAssignedIdentities@2023-01-31' = {
+  location: location
+  name: 'image-puller'
+}
+
+module acrPullerRoles 'acr/acr-permissions.bicep' = [
+  for (_, i) in acrPullResourceGroups: {
+    name: guid(acrRg[i].id, aksCluster.id, acrPullRoleDefinitionId)
```

this name definition is identical with the one used in `module acrPullRole` - hence the error during the PR check what-if
```suggestion
    name: guid(acrRg[i].id, aksCluster.id, acrPullRoleDefinitionId, 'puller-identity')
```